File: core/src/main/java/vn/loitp/core/utilities/LPopupMenu.java
package vn.loitp.core.utilities;

import android.app.Activity;
import android.view.MenuItem;
import android.view.View;
import android.widget.PopupMenu;

/**
 * Created by www.muathu@gmail.com on 5/13/2017.
 */

public class LPopupMenu {
    public interface CallBack {
        public void clickOnItem(MenuItem menuItem);
    }

    public static void show(final Activity activity, View showOnView, int menuRes, final CallBack callBack) {
        PopupMenu popup = new PopupMenu(activity, showOnView);
        popup.getMenuInflater().inflate(menuRes, popup.getMenu());
        popup.setOnMenuItemClickListener(new PopupMenu.OnMenuItemClickListener() {
            public boolean onMenuItemClick(MenuItem menuItem) {
                if (callBack != null) {
                    callBack.clickOnItem(menuItem);
                }
                return true;
            }
        });
        popup.show();
    }
}

File: core/src/main/java/com/core/utilities/LPopupMenu.kt
package com.core.utilities

import android.app.Activity
import android.view.MenuItem
import android.view.View
import android.widget.PopupMenu

/**
 * Created by www.muathu@gmail.com on 5/13/2017.
 */

class LPopupMenu {

    companion object {

        fun show(
                activity: Activity,
                showOnView: View,
                menuRes: Int,
                callback: ((MenuItem) -> Unit)? = null
        ) {
            val popup = PopupMenu(activity, showOnView)
            popup.menuInflater.inflate(menuRes, popup.menu)
            popup.setOnMenuItemClickListener { menuItem ->
                callback?.invoke(menuItem)
                true
            }
            popup.show()
        }
    }
}
